src/server/luaentity_sao.cpp

```diff
@@ -217,43 +216,7 @@ void LuaEntitySAO::step(float dtime, bool send_recommended)
 		}
 	}
 
-	if (!m_armor_groups_sent) {
-		m_armor_groups_sent = true;
-		// create message and add to list
-		m_messages_out.emplace(getId(), true, generateUpdateArmorGroupsCommand());
-	}
-
-	if (!m_animation_sent) {
-		m_animation_sent = true;
-		std::string str = generateUpdateAnimationCommand();
-		// create message and add to list
-		m_messages_out.emplace(getId(), true, str);
-	}
-
-	if (!m_animation_speed_sent) {
-		m_animation_speed_sent = true;
-		std::string str = generateUpdateAnimationSpeedCommand();
-		// create message and add to list
-		m_messages_out.emplace(getId(), true, str);
-	}
-
-	if (!m_bone_position_sent) {
-		m_bone_position_sent = true;
-		for (std::unordered_map<std::string, core::vector2d<v3f>>::const_iterator
-				ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii){
-			std::string str = generateUpdateBonePositionCommand((*ii).first,
-					(*ii).second.X, (*ii).second.Y);
-			// create message and add to list
-			m_messages_out.emplace(getId(), true, str);
-		}
-	}
-
-	if (!m_attachment_sent) {
-		m_attachment_sent = true;
-		std::string str = generateUpdateAttachmentCommand();
-		// create message and add to list
-		m_messages_out.emplace(getId(), true, str);
-	}
+	sendModificationMessages();
```

Comment:
I kinda don't like the name, maybe instead move the code to `UnitSAO::step()`?

Reply:
What name would you prefer? `sendDirtyMessages`, `sendOutdatedData`, ...

Whereas a `step` function would be totally feasible, I don't think this "skeleton" class should contain "self-living" functionalities, but let PlayerSAO/LuaEntitySAO call its functions wherever needed.

Reply:
Right. `sendOutdatedData` is fine by me.